test/spec/command/CommandStackSpec.js

```diff
@@ -876,4 +877,99 @@ describe('command/CommandStack', function() {
 
   });
 
+  describe('change-event', function() {
+
+    var testSetup = function(eventBus, commandStack) {
+      var eventData = {};
+
+      commandStack.registerHandler('complex-command', ComplexCommand);
+      commandStack.registerHandler('pre-command', PreCommand);
+      commandStack.registerHandler('post-command', PostCommand);
+      eventBus.on('commandStack.changed', function(event) {
+        eventData.changeEvent = event.method;
+        eventData.actions = event.actions.map(function(action) {return action.command;});
+      });
+
+      return eventData;
+    };
+
+    describe('should indicate details about a change', function() {
+
+      it('with method <execute>', inject(function(eventBus, commandStack) {
+
+        // given
+        var eventData = testSetup(eventBus, commandStack);
+
+        // when
+        commandStack.execute('complex-command', { element: { trace: [] } });
+
+        // then
+        var changeEvent = eventData.changeEvent,
+            actions = eventData.actions;
+
+        expect(changeEvent).to.equal('execute');
+        expect(actions).to.eql(['pre-command', 'complex-command', 'post-command']);
+
+      }));
+
+      it('with method <undo>', inject(function(eventBus, commandStack) {
+
+        // given
+        var eventData = testSetup(eventBus, commandStack);
+
+        // when
+        commandStack.execute('complex-command', { element: { trace: [] } });
+        commandStack.undo();
+
+        // then
+        var changeEvent = eventData.changeEvent,
+            actions = eventData.actions;
+
+        expect(changeEvent).to.equal('undo');
+        expect(actions).to.eql(['post-command', 'complex-command', 'pre-command']);
+
+      }));
+
+      it.skip('with method <redo>', inject(function(eventBus, commandStack) {
+
+        // given
+        var eventData = testSetup(eventBus, commandStack);
+
+        // when
+        commandStack.execute('complex-command', { element: { trace: [] } });
+        commandStack.undo();
+        commandStack.redo();
+
+        // then
+        var changeEvent = eventData.changeEvent,
+            actions = eventData.actions;
+
+        expect(changeEvent).to.equal('redo');
+
+        // Need to figure out a way to indicate the executed actions
+        expect(actions).to.eql(['pre-command', 'complex-command', 'post-command']);
```

Comment:
This one has an issue though, that I discovered when making tests (I skipped the relevant test because it fails). The redo action doesn't indicate the executed actions. I can't figure out a graceful way to do this, so I would appreciate a nudge in the right direction!